chemfiles/atom.py

```diff
@@ -19,121 +19,153 @@ class Atom(CxxPointer):
-            lambda buffer, size: self.ffi.chfl_atom_type(self, buffer, size),
-            initial=32,
+            lambda buffer, size: self.ffi.chfl_atom_type(self, buffer, size), initial=32
         )
 
-    def set_type(self, type):
-        '''Set this :py:class:`Atom` type to ``type``.'''
+    @type.setter
+    def type(self, type):
+        """Set this :py:class:`Atom` type to ``type``."""
         self.ffi.chfl_atom_set_type(self, type.encode("utf8"))
 
+    @property
     def full_name(self):
-        '''
+        """
         Try to get the full name of this :py:class:`Atom` from its type. For
         example, the full name of "He" is "Helium". If the name can not be
         found, returns the empty string.
-        '''
+        """
         return _call_with_growing_buffer(
             lambda buff, size: self.ffi.chfl_atom_full_name(self, buff, size),
             initial=64,
         )
 
+    @property
     def vdw_radius(self):
-        '''
+        """
         Try to get the Van der Waals radius of this :py:class:`Atom` from its
         type. If the radius can not be found, returns 0.
-        '''
+        """
         radius = c_double()
         self.ffi.chfl_atom_vdw_radius(self, radius)
         return radius.value
 
+    @property
     def covalent_radius(self):
-        '''
+        """
         Try to get the covalent radius of this :py:class:`Atom` from its type.
         If the radius can not be found, returns 0.
-        '''
+        """
         radius = c_double()
         self.ffi.chfl_atom_covalent_radius(self, radius)
         return radius.value
 
+    @property
     def atomic_number(self):
-        '''
+        """
         Try to get the atomic number of this :py:class:`Atom` from its type. If
         the atomic number can not be found, returns 0.
-        '''
+        """
         number = c_uint64()
         self.ffi.chfl_atom_atomic_number(self, number)
         return number.value
 
-    def set(self, name, value):
-        '''
-        Set a property of this atom, with the given ``name`` and ``value``.
-        The new value overwrite any pre-existing property with the same name.
-        '''
-        self.ffi.chfl_atom_set_property(
-            self, name.encode("utf8"), Property(value)
-        )
+    def __iter__(self):
+        # Disable automatic iteration from __getitem__
+        raise TypeError("can not iterate over an atom")
 
-    def get(self, name):
-        '''
+    def __getitem__(self, name):
+        """
         Get a property of this atom with the given ``name``, or raise an error
         if the property does not exists.
-        '''
+        """
+        if not isinstance(name, string_type):
+            raise ChemfilesError(
+                "Invalid type {} for an atomic property name".format(type(name))
+            )
         ptr = self.ffi.chfl_atom_get_property(self, name.encode("utf8"))
         return Property.from_ptr(ptr).get()
+
+    def __setitem__(self, name, value):
```

Comment:
Other special functions like `__repl__` may also be useful here. I typically use Python to prototype something before rewriting the prototype into a different language and this kind of support can be useful.

Reply:
Are you talking about `__repr__` ? If so yes, we could add them. I don't know how to format/represent an atom though.

Reply:
I'll do that in a separate PR/Issue